spec/signed_files.md

```diff
@@ -0,0 +1,78 @@
+## Signed Files
+
+`did:webs` [[ref: signed files]] combine the semantics of folders of files on
+web servers, and the [DID Core Specification]'s [DID URL path] semantics to
+enable the controller of the DID to publish documents cryptographically signed
+using the verification key(s) of the DID. This kind of signed file capability
+has been implemented in a variety of ways in the past, such as with [hashlinks].
+The value of including this as part of the `did:webs` DID method is that it
+ensures consistency in its application, including how the [[ref: signed files]]
+can be verified when using a [[ref: multisig]]-enabled DID, and in the face of
+rotations of the DID's key(s).
+
+[DID Core Specification]: https://www.w3.org/TR/did-core/
+[DID URL path]: https://www.w3.org/TR/did-core/#path
+[hashlinks]: https://datatracker.ietf.org/doc/html/draft-sporny-hashlink
+
+`did:webs` [[ref: signed files]] work as follows:
+
+- The controller of the DID may place arbitrary files into the folder structure
+  below the root did:webs folder.
+  - The DID URL for the [[ref: signed file]] is of the form `<did:webs DID>/<path
+  to file>`.
+  - The HTTPS path to the file MUST be `<HTTPS URL conversion of did:webs
+  identifier>/<path to file>.
+- To be a valid `did:webs` [[ref: signed file]], each file MUST have an
+  associated JSON Web Signature file named `<filename>.jws` beside (in the same Web folder as) the file.
+  - For example, a file `revocation_registry.bin` MUST have a file
+    `revocation_registry.bin.jws` beside it.
+  - The HTTPS path to the associated JWS file MUST be `<HTTPS URL conversion of
+  did:webs identifier>/<path to file>.jws`
+- The JWS web signature MUST:
+  - Have a payload that is the [multiformat multihash] hash of the content of the associated signed file.
+  - Be signed with a verification key(s) that is (are) either currently in the DIDDoc
+    of the DID, or found in the [[ref: KEL]] associated with the DID. A verification
```

Comment:
I'd say the "kid" header of the JWS needs to explicitly reference (via DID URL) the verification method that can be used to verify the JWS signature(s).

Reply:
Agreed.  The “or” is if the key has been rotated, and an older key, no longer in the DIDDoc is required.